### PR TITLE
Fix for #12004issue:Rewrite handleStringData to use CompositeIdFetcher

### PR DESCRIPTION
**Pull request by Red12138-java**
Refactor the handleStringData method as requirements and add a no-args constructor to CompositeFetcher.java, the reason to add no-args constructor is that performSearchById is not a static method which could not be used directly.

"Closes https://github.com/JabRef/jabref/issues/12004".

### Mandatory checks

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [X] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java [hunk 1]
```diff
@@ -19,6 +19,10 @@ public CompositeIdFetcher(ImportFormatPreferences importFormatPreferences) {
         this.importFormatPreferences = importFormatPreferences;
     }
 
+    public CompositeIdFetcher() {
+        importFormatPreferences = null;
```
Comment (koppor) [inline src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java:23]: No, you need to pass the preferences. - You have it availble in line 101 in ImportHandler.  Store it in a class variable and use the class variable in line 367 of ImportHandler.

**Review by koppor — CHANGES_REQUESTED** [review]

File: src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java (hunk 1 above)
Comment (Red12138-java) [reply src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java:23]: Oh, sorry. I see. I have deleted the constructor. I will create a pull request again. Thank you so much.

Comment (koppor) [reply src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java]: No additional pull request. Just push. As your teammate already did!

Comment (Red12138-java) [reply src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java]: Okkk!thank you

**Review by koppor — APPROVED** [review]
Tested. Works. Thank you for the PR.

Since this was really a quick-win, maybe you can tackle another issue? Now you can be more relaxed and really take this as opportunity to level-up your code reading and code writing skills.

We offer a curated list of issues at https://github.com/orgs/JabRef/projects/3/views/3.

**Pull request merged**